### PR TITLE
refactors formfield.border.error to formfield.error.border

### PR DESCRIPTION
**Pull request by mcraepv**
Signed-off-by: mcraepv <61661042+mcraepv@users.noreply.github.com>

#### What does this PR do?
Refactors FormField.border.error to formfield.error.border
#### Where should the reviewer start?
grommet\src\js\components\Form\stories\FieldStatesCustom.js
#### What testing has been done on this PR?
Storybook and pre-commit test suite
#### How should this be manually tested?
storybook @ /story/form--field-states-custom
#### What are the relevant issues?
#4004
#### Do the grommet docs need to be updated?
formField.border.error.color should be changed to formField.error.border.color

### CODE REVIEW

File: src/js/components/FormField/FormField.js [hunk 1]
```diff
@@ -242,8 +242,12 @@ const FormField = forwardRef(
       formFieldTheme.disabled.border.color
     ) {
       borderColor = formFieldTheme.disabled.border.color;
-    } else if (error && themeBorder && themeBorder.error.color) {
-      borderColor = themeBorder.error.color || 'status-critical';
+    } else if (
+      error &&
+      formFieldTheme.error.border &&
+      formFieldTheme.error.border.color
```
Comment (ShimiSun) [inline src/js/components/FormField/FormField.js:248]: For backward compatibility reasons, we should still support `themeBorder.error.color` if provided, and ignore it if `formFieldTheme.error.border.color` is defined.

File: src/js/themes/base.js [hunk 2]
```diff
@@ -736,12 +736,12 @@ export const generate = (baseSpacing = 24, scale = 6) => {
     formField: {
       border: {
         color: 'border',
-        error: {
-          color: {
-            dark: 'white',
-            light: 'status-critical',
-          },
-        },
+        // error: {
```
Comment (ShimiSun) [inline src/js/themes/base.js:739]: Can we add a comment that this will be deprectaed and one should use  `error.border` instead?

**Review by ShimiSun — COMMENTED** [review]
Thank you for this PR, dropped a few comments!

File: src/js/components/FormField/FormField.js (hunk 1 above)
Comment (mcraepv) [reply src/js/components/FormField/FormField.js:248]: Ok, I will update that tomorrow.

File: src/js/themes/base.js (hunk 2 above)
Comment (mcraepv) [reply src/js/themes/base.js:739]: Yes, I can add that. Thank you for the comments!

File: src/js/components/FormField/FormField.js [hunk 3]
```diff
@@ -242,8 +242,16 @@ const FormField = forwardRef(
       formFieldTheme.disabled.border.color
     ) {
       borderColor = formFieldTheme.disabled.border.color;
-    } else if (error && themeBorder && themeBorder.error.color) {
-      borderColor = themeBorder.error.color || 'status-critical';
+    } else if (
+      (error &&
+        formFieldTheme.error.border &&
+        formFieldTheme.error.border.color) ||
+      (error && themeBorder && themeBorder.error.color)
```
Comment (ShimiSun) [inline src/js/components/FormField/FormField.js:249]:
```suggestion
      (error && themeBorder && themeBorder.error.color) // backward compatibility check
```

**Review by jcfilben — APPROVED** [review]

**Review by jcfilben — COMMENTED** [review]
I accidentally hit approve instead of comment, disregard that. Code changes look good, but some of the checks for lint and jest are failing. Try merging your branch with master and make sure lint and jest are passing locally and then push the changes

**Review by jcfilben — COMMENTED** [review]
@ShimiSun I am not sure what is causing the snapshot failures on this PR. I looked over the code and I am not sure what is adding the following to three of the snapshots:
```
+ @media only screen and (max-width:768px) {
+
+ }
+
```
Is it safe to update the snapshots or should this PR not be causing any snapshot changes?

**Comment by IanKBovard** [comment]
This isn't backwards compatible. Tested with a custom theme.
```
formField: {
  border: {
    error: {
      color: 'blue'
    }
  }
}
```
the formField.error.border.color values in the base theme will be defaulted to. 

More logic needs to applied here. Something like this to cover most cases.

```
if (formField.border.error && ! formField.error.border) // use old theming styles and defaults
if(formField.border.error && formField.error.border) // use new theming styles and defaults
if(!formField.border.error && formField.error.border) // use new theming styled and defaults
```

**Comment by ShimiSun** [comment]
@mcraepv hi buddy, any updates on this PR?

**Comment by mcraepv** [comment]
Hey Shimi, so sorry for the delay. I'll have something for y'all this weekend.

**Comment by ShimiSun** [comment]
Hi buddy, I'll be closing this PR since it's hanging in there for a while. Feel free to address comments and conflicts and reopen if it's still relevant, Thank you!

**Pull request closed without merging**